src/poly/trans.lisp

```diff
@@ -1,14 +1,24 @@
 (in-package :geb.poly.trans)
 
-(defgeneric to-vampir (morphism value)
+(defgeneric to-vampir (morphism value let-vars)
   (:documentation "Turns a POLY term into a Vamp-IR term with a given value"))
 
 (defun to-circuit (morphism name)
   "Turns a POLY term into a Vamp-IR Gate with the given name"
-  (let ((wire (vamp:make-wire :var :x)))
-    (vamp:make-alias :name name
-                     :inputs (list wire)
-                     :body (list (to-vampir morphism wire)))))
+  (labels ((make-alias (name morphism)
+             (let ((wire (vamp:make-wire :var :x)))
+               (multiple-value-bind (results lets) (to-vampir morphism wire nil)
+                 (vamp:make-alias :name name
+                                  :inputs (list wire)
+                                  :body (reverse (cons results lets)))))))
+    (multiple-value-bind (morphism map) (ext:common-sub-expressions morphism)
```

Comment:
I think this is the exact one place where the compilation pipeline is currently calling the sub-expression module; is that right?

Reply:
Yeah, we can use it in other places, like in the graph, I could maybe use this so we can see what we can name our common patterns. Should be a good heuristic if we use it elsewhere